pkg/serverless/trace/inferredspan/span_enrichment_test.go

```diff
@@ -697,14 +698,197 @@ func TestRemapsSpecificInferredSpanServiceNamesFromEventBridgeEvent(t *testing.T
 	assert.Equal(t, "eventbridge", span2.Service)
 }
 
+func TestExtractContextFromSNSSQSEvent_ValidBinaryTraceData(t *testing.T) {
+	mockSQSMessage := events.SQSMessage{
+		Body: `{
+            "Message": "mock message",
+            "MessageAttributes": {
+                "_datadog": {
+                    "Type": "Binary",
+                    "Value": "eyJ0cmFjZXBhcmVudCI6IjAwLTAwMDAwMDAwMDAwMDAwMDAxN2ZlNGQ1ODA0YWMxNzg3LTA0ZThhY2JmZGY2YWE5OTktMDEiLCJ0cmFjZXN0YXRlIjoiZGQ9czoxO3QuZG06LTEiLCJ4LWRhdGFkb2ctdHJhY2UtaWQiOiIxNzI4OTA0MzQ3Mzg3Njk3MDMxIiwieC1kYXRhZG9nLXBhcmVudC1pZCI6IjM1MzcyMjUxMDgzNTYyNDM0NSIsIngtZGF0YWRvZy1zYW1wbGluZy1wcmlvcml0eSI6IjEiLCJ4LWRhdGFkb2ctdGFncyI6Il9kZC5wLmRtPS0xIn0="
```

Comment:
This b64 vdecoded is

{
  "traceparent": "00-000000000000000017fe4d5804ac1787-04e8acbfd6aa999-01",
  "tracestate": "dd=s:1;t.dm=-1",
  "x-datadog-trace-id": "1728904347387697031",
  "x-datadog-parent-id": "353722510835624345",
  "x-datadog-sampling-priority": "1",
  "x-datadog-tags": "_dd.p.dm=-1"
}